cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -1615,32 +1625,36 @@ def token_generator(length=32):
 
 
 @retry(times=3, delay_secs=10)
-def all_kube_system_pods_running():
-    ''' Check pod status in the kube-system namespace. Returns True if all
-    pods are running, False otherwise. '''
-    cmd = ['kubectl', 'get', 'po', '-n', 'kube-system', '-o', 'json']
+def get_kube_system_pods_not_running():
```

Comment:
The truthiness of this function got inverted (except for the `None` case), and that changes the behavior of the [@retry decorator](https://github.com/juju-solutions/kubernetes/blob/1123a8d10dec2136593432cdc12da99e2887da0e/cluster/juju/layers/kubernetes-common/lib/charms/layer/kubernetes_common.py#L58).

Before:

| Condition | Return value | Retry? |
| --- | --- | --- |
| Failed to get kube-system pod status | False | Yes |
| There are pods not running | False | Yes |
| All pods are running | True | No |

After:

| Condition | Return value | Retry? |
| --- | --- | --- |
| Failed to get kube-system pod status | None | Yes |
| There are pods not running | [...] | No |
| All pods are running | [] | Yes |

Reply:
what would you suggest here? I see a few options:
1) No retry decorator. I assume the goal was to hope that things would sort themselves out quickly and to wait for it, but is this valuable? I can see arguments both ways.
2) change function around so the retry handler works as expected, but then the function would return odd things based on name. I assume this would require changing the function name to something that matched the return values.
3) Bail on the idea that the user should know how many pods are not working.